### PR TITLE
Implemented signOut function for SettingsFragment

### PR DESCRIPTION
**Pull request by nervehammer**
## Description
This PR adds functionality to the SettingsFragment's Sign Out option

## Related Issue

Fixes: Issue #95 
## Motivation and Context
To enable user, to be able to sign out from the application

## How Has This Been Tested?
Rigorously tested after compiling the app

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds a new functionality)
- [ ] Code cleanup (non-breaking change which improves the code structure)
- [ ] Breaking change (fix or a feature that would cause the existing functionality to change) 

## Checklist:

- [ ] My code follows the code style of this project.
- [x] There is a corresponding issue for this pull request.
- [ ] I have updated the documentation (if required).

### CODE REVIEW

File: app/src/main/java/com/procleus/brime/ui/SettingsActivity.java [hunk 1]
```diff
@@ -69,6 +72,19 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     }
 
+    public void signOut(){
+        SharedPreferences sharedpreferences = getSharedPreferences(SigninActivity.PREF, Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedpreferences.edit();
+        editor.remove("emailpref");
```
Comment (imujjwal96) [inline app/src/main/java/com/procleus/brime/ui/SettingsActivity.java]: can we not remove everything from shared pref in a single statement?

Comment (syedsaadh) [inline app/src/main/java/com/procleus/brime/ui/SettingsActivity.java]: I would highly recommend to make a `AccountUtils` class which handles all the account related actions with the shared preferences.
for example 
```
public class AccountUtils {
    private static final String TAG = AccountUtils.class.getSimpleName();

    public static final String PREF_ACTIVE_ACCOUNT = "chosen_account";
    public static final String PREF_ACTIVE_ACCOUNT_NAME = "chosen_account_name";
    public static final String PREF_ACTIVE_ACCOUNT_EMAIL = "chosen_account_email";
   
    private static SharedPreferences getSharedPreferences(final Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    /**
     * Specify whether the app has an active account set.
     *
     * @param context Context used to lookup {@link SharedPreferences} the value is stored with.
     */

    public static boolean hasActiveAccount(final Context context) {
        return !TextUtils.isEmpty(getActiveAccountEmail(context));
    }
   public static String getActiveAccountEmail(final Context context) {
        SharedPreferences sp = getSharedPreferences(context);
        return sp.getString(PREF_ACTIVE_ACCOUNT, null);
    }
    public static void setActiveAccount(final Context context, final String accountEmail) {
        SharedPreferences sp = getSharedPreferences(context);
        sp.edit().putString(PREF_ACTIVE_ACCOUNT, accountEmail).apply();
    }

    public static void clearActiveAccount(final Context context) {
        Log.d(TAG, "Clearing active account");
        SharedPreferences sp = getSharedPreferences(context);
        sp.edit().remove(PREF_ACTIVE_ACCOUNT).apply();
        sp.edit().remove(PREF_ACTIVE_ACCOUNT_NAME).apply();
        sp.edit().remove(PREF_ACTIVE_ACCOUNT_REFERRAL_CODE).apply();
    }
}
```

**Review by syedsaadh — CHANGES_REQUESTED** [review]
@nervehammer If you could do suggested changes then that would be perfect.

**Comment by nervehammer** [comment]
@syedsaadh o.k i will try to implement them according to requested changes :D

**Pull request closed without merging**